lib/tweet.rb

```diff
@@ -26,6 +25,17 @@ def initialize(options={})
 
   def save
     ## Fire a sql statement to create a new tweet
+    ## Should create a new tweet if it doesn't exist yet
+    if persisted?
+      # update the tweet instance in the table
+      update_tweet_message_in_database
```

Comment:
Nice - I'd probably name this method just `update` to be a little bit less wordy, but that's just personal preference.